cw_lightning/lib/lightning_wallet.dart

```diff
@@ -0,0 +1,407 @@
+    required Box<UnspentCoinsInfo> unspentCoinsInfo,
+    required String password,
+  }) async {
+    final snp =
+        await ElectrumWalletSnapshot.load(name, walletInfo.type, password, BitcoinNetwork.mainnet);
+    return LightningWallet(
+      mnemonic: snp.mnemonic!,
+      password: password,
+      walletInfo: walletInfo,
+      unspentCoinsInfo: unspentCoinsInfo,
+      initialAddresses: snp.addresses,
+      initialBalance: LightningBalance(
+        confirmed: snp.balance.confirmed,
+        unconfirmed: snp.balance.unconfirmed,
+        frozen: snp.balance.frozen,
+      ),
+      seedBytes: await Mnemonic.toSeed(snp.mnemonic!),
+      initialRegularAddressIndex: snp.regularAddressIndex,
+      initialChangeAddressIndex: snp.changeAddressIndex,
+      addressPageType: snp.addressPageType,
+    );
+  }
+
+  Future<void> _handleNodeState(NodeState? nodeState) async {
+    if (nodeState == null) return;
+    _balance[CryptoCurrency.btcln] = LightningBalance(
+      confirmed: nodeState.maxPayableMsat ~/ 1000,
+      unconfirmed: nodeState.maxReceivableMsat ~/ 1000,
+      frozen: 0,
+    );
+  }
+
+  Future<void> _handlePayments(List<Payment> payments) async {
+    _isTransactionUpdating = true;
+    final txs = convertToTxInfo(payments);
+    transactionHistory.addMany(txs);
+    _isTransactionUpdating = false;
+  }
+
+  @override
+  Future<void> renameWalletFiles(String newWalletName) async {
+    await stopBreez(true);
+    await super.renameWalletFiles(newWalletName);
+    await setupBreez(await Mnemonic.toSeed(mnemonic));
+  }
+
+  Future<void> setupBreez(Uint8List seedBytes) async {
+    final sdk = await BreezSDK();
+    try {
+      if (!(await sdk.isInitialized())) {
+        sdk.initialize();
+      }
+    } catch (e) {
+      print("Error initializing Breez: $e");
+      return;
+    }
+
+    // sdk.logStream.listen((LogEntry event) {
+    //   print("Breez log: ${event.line}");
+    // });
+
+    Uint8List deviceKey = base64.decode(secrets.greenlightKey);
+    Uint8List deviceCert = base64.decode(secrets.greenlightCert);
+    GreenlightCredentials greenlightCredentials = GreenlightCredentials(
+      deviceKey: deviceKey,
+      deviceCert: deviceCert,
+    );
+
+    NodeConfig breezNodeConfig = NodeConfig.greenlight(
+      config: GreenlightNodeConfig(
+        partnerCredentials: greenlightCredentials,
+        inviteCode: null,
+      ),
+    );
+    Config breezConfig = await sdk.defaultConfig(
+      envType: EnvironmentType.Production,
+      apiKey: secrets.breezApiKey,
+      nodeConfig: breezNodeConfig,
+    );
+
+    String workingDir = await pathForWalletDir(name: walletInfo.name, type: type);
+    workingDir = "$workingDir/breez/";
+
+    new Directory(workingDir).createSync(recursive: true);
+    breezConfig = breezConfig.copyWith(workingDir: workingDir);
+
+    // disconnect if already connected
+    try {
+      if (await sdk.isInitialized()) {
+        await sdk.disconnect();
+      }
```

Comment:
so at the start of the function you initialize it, and here you disconnect it?! is it just to get the `defaultConfig` or what?

Reply:
this might no longer be strictly necessary, but should be safe to leave in as disconnect does nothing if already disconnected (where as calling connect when already connected fails and I think even kills the existing connection)